src/js/tabs/auxiliary.js

```diff
@@ -13,8 +13,222 @@ import inflection from "inflection";
+
+// Categories to be mapped with buildMap. Category 'all' are virtuel and always included
+const categoryTable = [
+    { name: 'BEEP',           buildKey: ['all'],       modes: ['BEEPERON', 'BEEPER', 'BEEPER MUTE', 'GPS BEEP SATELLITE COUNT']},
+    { name: 'BLACKBOX',       buildKey: ['all'],       modes: ['BLACKBOX', 'BLACKBOX ERASE']},
+    { name: 'CAM',            buildKey: ['cam'],       modes: ['CAMERA CONTROL 1', 'CAMERA CONTROL 2', 'CAMERA CONTROL 3']},
+    { name: 'DSHOT / 3D',     buildKey: ['dshot'],     modes: ['3D', '3D DISABLE / SWITCH', 'FLIP OVER AFTER CRASH']},
+    { name: 'FLIGHT CONTROL', buildKey: ['all'],       modes: flightControl},
+    { name: 'GPS / MAG',      buildKey: ['gps_mag'],   modes: ['GPS BEEP SATELLITE COUNT', 'GPS RESCUE', 'MAG']},
+    { name: 'LED',            buildKey: ['led_strip'], modes: ['LEDLOW']},
+    { name: 'OSD',            buildKey: ['osd'],       modes: ['OSD DISABLE']},
+    { name: 'OTHER',          buildKey: ['all'],       modes: ['CALIB', 'MSP OVERRIDE', 'LAP TIMER RESET', 'PASSTHRU', 'PARALYZE', 'PID AUDIO', 'PREARM', 'READY']},
+    { name: 'SERVO',          buildKey: ['servos'],    modes: ['SERVO1', 'SERVO2', 'SERVO3']},
+    { name: 'TELEMETRY',      buildKey: ['telemetry'], modes: ['TELEMETRY']},
+    { name: 'USER',           buildKey: ['all'],       modes: ['USER1', 'USER2', 'USER3', 'USER4']},
+    { name: 'VTX',            buildKey: ['vtx'],       modes: ['STICK COMMANDS DISABLE', 'VTX CONTROL DISABLE', 'VTX PIT MODE']},
+];
+
+let hideUnusedModes = false;
+let modeList = [];
+
+function inBuildMap(map, name) {
+    if (name == 'all') {
+        return true;
+    }
+    for (let value of map) {
+        if (name.includes(value.buildKey)) {
+            for (let option of value.buildOption) {
+                if (FC.CONFIG.buildOptions.includes(option)) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+function isSelectedMode(mList, modeName) {
+    for (let i = 0; i < mList.length; i++) {        //    for (let value of mList) don't break on first hit
+        if (mList[i].includes(modeName)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function resolveCategoryName(category, choise) {
+    let mList = [];
+    for (let value of choise) {
+        for (let elm of category) {
+            if (value.includes(elm.name)) {
+                mList.push(elm.modes);
+            }
+        }
+    }
+    return mList;
+}
+
+function updateSearchResults() {
+    let categorySelect = $('select.auxiliary_category_select');
+
+    const categoryNameList = categorySelect.multipleSelect("getSelects", "text");
+    setConfig({ auxiliaryCategoryNameList: categoryNameList });         // save as users choise
+    modeList = resolveCategoryName(categoryTable, categoryNameList);
+    updateModes();                                                      // update UI
+}
+
+function getCategoryNames(table, buildKey) {
```

Comment:
There's a typo: "choice". Also I don't get why passing the table is necessary once it is already in scope by the name `categoryTable`. It is just confusing